### PR TITLE
v2.0.0 Refactors

### PR DESCRIPTION
**Pull request by bearror**
_Tracking v2.0.0 refactors in this PR._

I've decided to **drop off Node 9 support**, as doing that this early on seems like the right call. This'll make Oletus **target Node 12 going forward**.

This also marks a good place for me taking Oletus a little more seriously. I'm going over the repository and cleaning up the fast and ugly code left behind. The refactored code will also include JSDoc annotations with type information for type checking during development. (Pending TypeScript support for .mjs input files for this to be truly useful.)

---

Further internal refactors will follow in the coming weeks. Expect minor changes to the reporter output, but no breaking changes that affect systems using Oletus are expected in the near future.

### CODE REVIEW

**Comment by bearror** [comment]
@Avaq if you're in a particular hurry, I don't mind getting a version out with the current changes, as it's enough to pass the tests as-is.

My current plan is to go through `runner` (probably removing `report`), cleaning up both reporters (probably moving the ugly formatting stuff into proper `util` functions), and then quickly check the rest.

**Comment by Avaq** [comment]
> I don't mind getting a version out with the current changes

Could you, please? I have several PR's waiting for a release of the changes recently made to Oletus.

File: util.mjs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+const cwd = `file://${process.cwd()}`
```
Comment (Avaq) [inline util.mjs]: Don't forget to add this to the `files` section in `package.json`

Comment (bearror) [reply util.mjs]: Good catch, fixing.

**Comment by bearror** [comment]
Thanks for the patience, I'm merging the current changes -- including the `cwd` fix. Went with straight up Node 12 support for now, as I'm still considering how the new implementation will affect Oletus. This way I can ensure there won't be a breaking change to this repository next month.

**Comment by bearror** [comment]
@Avaq anything else I should address now before bumping to 2.0.0?

**Comment by Avaq** [comment]
> This way I can ensure there won't be a breaking change to this repository next month.

See https://github.com/nodejs/modules/blob/master/doc/plan-for-new-modules-implementation.md - the module system as it is in Node 12 is unlikely to change anymore.

> anything else I should address now before bumping to 2.0.0?

Nothing off the top of my head. I've been building on top of the current `#master` without any problems.

**Comment by Avaq** [comment]
I'll try this branch to see if I run into anything.

**Comment by bearror** [comment]
Alright, great.

I'll experiment with dropping the `mjs` extension in the future for better TypeScript support now that Oletus isn't tied to the <12 module system. Like stated above, though, I'm not going to make changes that leak out of the package.

**Comment by bearror** [comment]
Merging and bumping to 2.0.0. It'll be on NPM in a couple of minutes. I've got time this weekend so if you come across any problems I'll address the issues right away.

**Pull request merged**